manager/controlapi/common.go

```diff
@@ -9,6 +10,8 @@ import (
 	"google.golang.org/grpc/codes"
 )
 
+var isValidName = regexp.MustCompile("\\A[a-zA-Z0-9](?:(?:[a-zA-Z0-9-]*[a-zA-Z0-9])|[a-zA-Z0-9]*)\\z")
```

Comment:
`^[A-Za-z0-9](?:[-_]*[A-Za-z0-9]+)*$'`

This should be good:

1. At least one character.
2. Allows embedded, consecutive `-` or `_`.